t/t2800-jobs-recursive.t

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+test_description='Test flux jobs command with --recursive'
+
+. $(dirname $0)/sharness.sh
+
+test_under_flux 2 job
+
+export FLUX_URI_RESOLVE_LOCAL=t
+runpty="${SHARNESS_TEST_SRCDIR}/scripts/runpty.py --line-buffer -f asciicast"
+waitfile="${SHARNESS_TEST_SRCDIR}/scripts/waitfile.lua"
+
+test_expect_success 'start a recursive job' '
+	id=$(flux mini submit flux start /bin/true) &&
+	rid=$(flux mini submit -n2 \
+		flux start \
+		flux mini submit --wait --cc=1-2 flux start \
+			"flux mini submit sleep inf && \
+                 	 touch ready.\$FLUX_JOB_CC && \
+		 	 flux queue idle") &&
+	flux job wait-event $id clean
+'
```

Comment:
maybe add a small comment here, takes a second to grok (i think 1 sub instance that is finished, 1 sub instance that has 2 sub instances within it).  Not sure of the purpose of the idle queue?

Reply:
`flux queue idle` waits for the running job to finish. Otherwise instance exits and job is killed.

Reply:
Good point, I'll add a comment before starting that test workload.